lib/core/uri_parser.js

```diff
@@ -106,6 +106,15 @@ function parseSrvConnectionString(uri, options, callback) {
         }
 
         record = qs.parse(record[0].join(''));
+
+        if (
+          Object.keys(record)
```

Comment:
below, in L118, it says that only two record keys are allowed, so does this just fail early in case the key is specifically "loadbalanced"? if so, would it not make more sense to put this logic in L119? just before the other error callback? or should this not be here at all since we already do the check in parsedOptions in L610?

Reply:
This is the options that are specified in the DNS TXT record so we do need to keep the check here, I figure that this LB check if statement is kept exactly the same in the 4 places I had to add it, to remain consistent for this exception.

I can take the more refactor-y approach where I can make the `Object.keys(record).some` check work on an array of keys, that's what we've done in 4.x, up until now there's only been two options permitted in DNS TXT record, now there's a third, maybe we should make this section more future proof.

Reply:
As a suggestion, if you use `Object.keys(...).some(k => k.toLowerCase() === 'loadbalanced')` instead of the map/indexOf combination, it'll look a bit cleaner; I think I can live with it being where it currently is in all 4 places if it's less of a visual sprawl 😆

Reply:
turned out to be easy enough to do, and yea the .some is much cleaner so I've used that in each place.